### ISSUE TITLE
Improve regex for keywords that can also be values

### ISSUE THREAD

**Issue by fladson** [fladson/vim-kitty#2]
For some keywords (`background`, `foreground`, etc.) that can also be values, we need to check if the string is at the beginning of the line or not to treat it accordingly.

We can remove the repetition in declaring those with a better regex expression.

**Comment by crater2150** [comment]
Hi, I started writing a [kitty syntax file](https://github.com/crater2150/nvim-config/blob/master/syntax/kitty.vim) this weekend before coming across this one. Maybe we could merge our versions?

I solved this issue by introducing an extra layer of matching around keywords:
```vim
syn match kittyKW '^\S*' contains=kittyOptions
syn match kittyMapKW '^\(mouse_\)\?map' contains=kittyMap
			\ nextgroup=kittyKeybind
```
`kittyOptions` is equivalent to your `kittyKeyword`, but without removing options, that may also be values. Using the `contained` parameter, they then are all only matched if at the beginning of the line.
If you'd like, I can create a PR.
